### PR TITLE
Added n_features_in_ to three of the embedded class estimators (continued from #434)

### LINKED ISSUE #352: Add `n_features_in_` for scikit-learn compliance
`FutureWarning: As of scikit-learn 0.23, estimators should expose a n_features_in_ attribute, unless the 'no_validation' tag is True. This attribute should be equal to the number of features passed to the fit method. An error will be raised from version 0.25 when calling check_estimator().`

### PR DESCRIPTION
**Pull request by zeou1**
#### Reference Issues/PRs

Fixes #352. See also PR #434 

#### What does this implement/fix? Explain your changes.
Added n_features_in_ to three embedded class estimators, ase.py, lse.py, and mds.py
For ase and lse, the n_features_in_ attribute equals to the number of points in the graph. And for mds, the attribute equals to the number of features of the first dimension of each sample.

Compared to PR #434, this pull request changes the definition of n_features_in_ for mds from the total number of features to the number of features in the first dimension.

#### Any other comments?
This pull request is made after the merge with Microsoft. All future conversation about #352 and #434 should be continued here.

### CODE REVIEW

**Comment by zeou1** [comment]
@asaadeldin11 Here is the new PR from #434

**Comment by asaadeldin11** [comment]
@zeou1 can you make some inconsequential commit and then revert it to trigger netlify?

**Review by asaadeldin11 — APPROVED** [review]
This looks good to me

**Review by dwaynepryce — APPROVED** [review]

**Comment by zeou1** [comment]
Should I commit again to trigger the new checks?

**Comment by dwaynepryce** [comment]
First you will need to merge `dev` back into your branch and handle any merge conflicts (which... may be substantial, given the number of changes lately)

File: graspy/embed/ase.py [hunk 1]
```diff
@@ -71,6 +71,8 @@ class AdjacencySpectralEmbed(BaseEmbed):
 
     Attributes
     ----------
+    n_features_in_: int
+        number of features passed to the fit method.
```
Comment (bdpedigo) [inline graspy/embed/ase.py:75]:
```suggestion
        Number of features passed to the fit method.
```

Comment (bdpedigo) [reply graspy/embed/ase.py:75]: @zeou1 please make this change everywhere and update

File: graspy/embed/ase.py [hunk 2]
```diff
@@ -154,5 +156,6 @@ def fit(self, graph, y=None):
         if self.diag_aug:
             A = augment_diagonal(A)
 
+        self.n_features_in_ = len(A)
```
Comment (bdpedigo) [inline graspy/embed/ase.py:159]: this can be refactored to `BaseEmbed` since it is common to both functions. same with the documentation

**Review by bdpedigo — CHANGES_REQUESTED** [review]

File: graspy/embed/base.py [hunk 3]
```diff
@@ -50,6 +50,8 @@ class BaseEmbed(BaseEstimator):
     ----------
     n_components_ : int
         Dimensionality of the embedded space.
+    n_features_in_: int
+        number of features passed to the fit method.
```
Comment (bdpedigo) [inline graspy/embed/base.py:54]:
```suggestion
        Number of features passed to the fit method.
```

File: graspy/embed/lse.py [hunk 4]
```diff
@@ -66,6 +66,9 @@ class LaplacianSpectralEmbed(BaseEmbed):
 
     Attributes
     ----------
+    n_features_in_: int
+        number of features passed to the fit method.
```
Comment (bdpedigo) [inline graspy/embed/lse.py:70]:
```suggestion
        Number of features passed to the fit method.
```

File: graspy/embed/mds.py [hunk 5]
```diff
@@ -60,6 +60,9 @@ class ClassicalMDS(BaseEstimator):
         Equals the parameter n_components. If input n_components was None,
         then equals the optimal embedding dimension.
 
+    n_features_in_: int
+        number of features passed to the fit method.
```
Comment (bdpedigo) [inline graspy/embed/mds.py:64]:
```suggestion
        Number of features passed to the fit method.
```

File: graspy/embed/ase.py (hunk 2 above)
Comment (bdpedigo) [reply graspy/embed/ase.py:159]: going to make a new issue for this for now

**Comment by zeou1** [comment]
Since a new issue is created for the redundant code in ase and lse, do I still need to fix that for this issue?

**Comment by bdpedigo** [comment]
let's just stick with the PR you have now, we are trying to get this merged in soon so I'd rather not modify it anymore

**Review by bdpedigo — APPROVED** [review]

**Pull request merged**